numpy/exceptions.py

```diff
@@ -195,3 +196,49 @@ def __str__(self):
             if self._msg is not None:
                 msg = f"{self._msg}: {msg}"
             return msg
+
+
+class InvalidPromotion(TypeError):
+    """Multiple DTypes could not be converted to a common one.
+
+    This exception derives from ``TypeError`` and is raised whenever dtypes
+    cannot be converted to a single common one.  This can be because they
+    are of a different category/class or incompatible instances of the same
+    one (see Examples).
+
+    Notes
+    -----
+    ``InvalidPromotion`` derives from ``TypeError`` and if it occurs within
```

Comment:
replace "if it occurs within" with "in"

Reply:
rephrasing a bit, the use of parentheses wasn't nice anyway.